script.rb

```diff
@@ -0,0 +1,148 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    until i == size
+      yield self[i]
+
+      i += 1
+    end
+    self
+  end
+
+  # my_each_with_index
+
+  def my_each_with_index
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    until i == size
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  # my_select
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    selected_array = []
+    i = 0
+    until i == size
+      selected_array << self[i] if yield(self[i])
+      i += 1
+    end
+    selected_array
+  end
+
+  # my_all
+
+  def my_all?(arg = nil)
+    if block_given?
+      my_each { |el| return false if yield(el) == false }
+      return true
+    elsif arg.nil?
+      my_each { |k| return false if k.nil? || k == false }
+    elsif !arg.nil? && (arg.is_a? Class)
+      my_each { |k| return false if k.class != arg }
+    elsif !arg.nil? && arg.instance_of?(Regexp)
+      my_each { |k| return false unless arg.match(k) }
+    else
+      my_each { |k| return false if k != arg }
+    end
+    true
+  end
```

Comment:
- [ ] `my_all` is not working as the original `all?` method also is not passing all test cases.

### `all?` is an Enumerable that has these characteristics:

- Returns `false` if it finds at least one element that is not true
```
[1, 'string', nil].all?
# false
```

- it can accept as arguments a Class, a RegEx, or a pattern, and returns true or false depending if the values in the array/range/hash match the condition
```
[1, 2, 'hello'].all?(Numeric) 
# false
```

[Enumerable#all?](https://ruby-doc.org/core-2.7.2/Enumerable.html#method-i-all-3F)